src/main/java/com/criticalsoftware/giveaway/Announcement.java

```diff
@@ -0,0 +1,91 @@
+package com.criticalsoftware.giveaway;
+
+import com.criticalsoftware.Product;
+import io.quarkus.mongodb.panache.PanacheMongoEntity;
+import org.bson.types.ObjectId;
+import java.time.LocalDateTime;
+
+public class Announcement extends PanacheMongoEntity {
+
+    private String id;
+    private Product product;
+    private LocalDateTime date;
+    private ObjectId userDonorId;
+    private ObjectId userDoneeId;
+    private boolean claimed;
+    private String claimedBy;
+
+    public Announcement() {}
+
+    public Announcement(String Id, Product product, ObjectId userDonorId, ObjectId userDoneeId, boolean claimed, String claimedBy) {
+        this.id = id;
+        this.product = product;
+        this.date = LocalDateTime.now();
+        this.userDonorId = userDonorId;
+        this.userDoneeId = userDoneeId;
+        this.claimed = claimed;
+        this.claimedBy = claimedBy;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    //Product
+    public Product getProduct(){
+        return this.product;
+    }
+
+    public void setProduct(Product product) {
+        this.product = product;
+    }
+
+    //Date
+    public LocalDateTime getdate(){
```

Comment:
Method names should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.